src/App.tsx

```diff
@@ -1,24 +1,227 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import * as todoService from './api/todos';
+import { Todo } from './types/Todo';
+import { TodosError } from './types/TodosError';
+import { Header } from './components/Header';
+import { TodoItem } from './components/TodoItem';
+import { TodosSortType } from './types/TodosSortType';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
 
-const USER_ID = 0;
+const filteredTodos = (todos: Todo[], filterStatus: TodosSortType): Todo[] => {
+  return todos.filter((todo: Todo) => {
+    switch (filterStatus) {
+      case TodosSortType.Completed:
+        return todo.completed;
+      case TodosSortType.Active:
+        return !todo.completed;
+      default:
+        return 1;
+    }
+  });
+};
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [todoTitle, setTodoTitle] = useState('');
+  const [todosSortBy, setTodosSortBy] = useState(TodosSortType.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [processingTodoIds, setProcessingTodoIds] = useState<number[]>([]);
+
+  const trimmedTitle = todoTitle.trim();
+
+  const visibleTodos = useMemo(() => (
+    filteredTodos(todos, todosSortBy)
+  ), [todos, todosSortBy]);
+
+  useEffect(() => {
+    todoService
+      .getTodos()
+      .then(setTodos)
+      .catch((error) => {
+        setErrorMessage(TodosError.DOWNLOAD_ERROR_MESSAGE);
+        throw error;
```

Comment:
```suggestion
        throw Error(error.message);
```
Fix this in all code.